### PR TITLE
Closes #1345 extends parameters of ranger learners 

### LINKED ISSUE #1345: ranger: new hyperpars available
I can see at least 3 new parameters:
- `splitrule`: Splitting rule, regression and survival only. For regression one of "variance" or "maxstat" with default "variance". For survival "logrank", "C" or "maxstat" with default "logrank".
- `alpha`: For "maxstat" splitrule: Significance threshold to allow splitting.
- `minprop`: For "maxstat" splitrule: Lower quantile of covariate distribtuion to be considered for splitting.
Maybe there are more, we should extend the param set.

### PR DESCRIPTION
**Pull request by florianfendt**
* I added the new parameters for the split rules
* ````write.forest = TRUE```` is now the default anyway so I adapted our code
* Added parameter ````keep.inbag````
* Extended tests to include new split rule parameters

### CODE REVIEW

File: R/RLearner_classif_ranger.R [hunk 1]
```diff
@@ -16,18 +16,19 @@ makeRLearner.classif.ranger = function() {
       makeUntypedLearnerParam(id = "always.split.variables"),
       makeLogicalLearnerParam(id = "respect.unordered.factors", default = FALSE),
       makeDiscreteLearnerParam(id = "importance", values = c("none", "impurity", "permutation"), default = "none", tunable = FALSE),
-      makeLogicalLearnerParam(id = "write.forest", default = FALSE, tunable = FALSE),
+      makeLogicalLearnerParam(id = "write.forest", default = TRUE, tunable = FALSE),
```
Comment (jakob-r) [inline R/RLearner_classif_ranger.R]: the default changed indeed

File: tests/testthat/test_surv_ranger.R [hunk 2]
```diff
@@ -1,33 +1,36 @@
 context("surv_ranger")
 
 ## TODO: Proper test required when predictions working
+## Suppress 'experimental' warning for splirule = maxstat
```
Comment (jakob-r) [inline tests/testthat/test_surv_ranger.R]: IMHO we should leave out the test for splitrule = "maxstat" as this is experimental.

**Comment by jakob-r** [comment]
Argh. I still don't get this review system completely. Can you remove the tests for splitrule == "maxstat"? If we want to test other parameter settings then just ntree there are a lot of other options.

**Comment by larskotthoff** [comment]
Thanks, merged.

**Pull request merged**